backend/algorithms/objects/conditions.py

```diff
@@ -239,7 +249,20 @@ def _validate_course(category: Category):
                 unlocked, warnings = list(zip(*validations))
                 satisfied = all(unlocked) if category.logic == Logic.AND else any(unlocked)
                 return satisfied, warnings
-            elif isinstance(category, ClassCategory):
+
+
+            course = course.class_name
+            if course not in user.courses:
+                return False, []
+
+            user_grade = user.get_grade(course)
+            if user_grade is None:
+                return True, [self.get_warning()]
+            if user_grade < self.grade:
+                return False, [f"Your grade {user_grade} in course {course} does not meet the grade requirements (minimum {self.grade}) for this course"]
+            return True, []
+
+            '''elif isinstance(category, ClassCategory):
```

Comment:
delete stuff if you dont want it, dont comment out.

Reply:
in the worst case, we can recover it using git

Reply:
yep will do, this was me being confused by the git conflict and not remembering what was going on here